### PR TITLE
Use the same versions file everywhere, pin to tf 1.0+ provider 4.0+

### PR DESCRIPTION
**Pull request by ludoo**

### CODE REVIEW

**Comment by ludoo** [comment]
One alternative is to just inject `versions.tf` during tests, tftest already supports it and we would only need to change the pytest fixtures we use for plan.

File: cloud-operations/dns-fine-grained-iam/versions.tf [hunk 1]
```diff
@@ -12,9 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-provider "google" {
-  impersonate_service_account = "data-platform-main@${var.project_ids.services}.iam.gserviceaccount.com"
```
Comment (sruffilli) [inline cloud-operations/dns-fine-grained-iam/versions.tf:16]: Is it ok to lose this?

File: cloud-operations/dns-fine-grained-iam/versions.tf [hunk 2]
```diff
@@ -12,9 +12,18 @@
...
+      version = ">= 4.0.0"
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = ">= 4.0.0"
+    }
+  }
 }
-provider "google-beta" {
-  impersonate_service_account = "data-platform-main@${var.project_ids.services}.iam.gserviceaccount.com"
```
Comment (sruffilli) [inline cloud-operations/dns-fine-grained-iam/versions.tf:19]: Same as above

File: cloud-operations/dns-fine-grained-iam/versions.tf (hunk 1 above)
Comment (ludoo) [reply cloud-operations/dns-fine-grained-iam/versions.tf:16]: ah no, good catch

**Comment by ludoo** [comment]
Fixes #354

**Review by juliocc — APPROVED** [review]

**Pull request merged**